conmon/conmon.c

```diff
@@ -618,7 +619,13 @@ static bool read_stdio(int fd, stdpipe_t pipe, bool *eof)
 
 static void on_sigchld(G_GNUC_UNUSED int signal)
 {
-	raise (SIGUSR1);
+	raise(SIGUSR1);
+}
+
+static void on_sig_exit(int signal)
+{
+	if (container_pid > 0)
+		kill(container_pid, signal);
```

Comment:
We probably also want to store `create_pid` in a global and signal it too.  That way we correctly handle the case where we receive a `TERM` after `fork`ing off the creation process but before we've read in the container PID.  There will still be a hole if we get a `TERM` [after the creation process exits but before we've read the PID file][1].  I'd rather not block signals that long, but can't think of a great way to cover that hole.  Maybe creating globals for `received_term`, `received_quit`, and `received_int` and then sending them off to the container PID once we have a PID?  In that case, the phases would be something like:

* Before we setup the handlers, `conmon` dies (with gentle cleanup?) on any of these signals.
* Once we fork the create process, we forward these signals to that process.
* Once the create process dies, attempts to forward the signals will error `ESRCH`.  Cache the calls in `received_term`, etc.
* Once we reap the create process, we stop attempting to signal the create process.  Continue to cache received signals in `received_term`, etc.
* Once we have `container_pid`, drain out any `received_term` and similar by forwarding them to the container.  Any new signals will be forwarded to `container_pid`.
* Once the container dies, attempts to forward the signals will error `ESRCH`.  Jump to our gentle cleanup.
* Once we reap the container process, we stop attempting to signal the container process.  Continue to jump to our gentle cleanup on new signals.

[1]: https://github.com/kubernetes-incubator/cri-o/blob/01b118116d3f188869a368d40561a2095aa57998/conmon/conmon.c#L1406-L1444